### PR TITLE
Update k8s dependencies to v0.24.2

### PR DESCRIPTION
**Pull request by aledbf**
## Description

Update go dependencies

## Release Notes
```release-note
NONE
```

## Werft options:

- [ ] /werft with-preview

### CODE REVIEW

**Comment by roboquat** [comment]
@aledbf: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Review by jenting — APPROVED** [review]

File: components/content-service/pkg/storage/gcloud_test.go [hunk 1]
```diff
@@ -66,230 +53,3 @@ func TestObjectAccessToNonExistentObj(t *testing.T) {
 		t.Errorf("gcloud storage reported object found despite it being non-existent")
 	}
 }
-
-var runWithDocker = flag.Bool("with-docker", false, "run fake-gcs-server in docker")
-
-func TestObjectUpload(t *testing.T) {
```
Comment (felladrin) [inline components/content-service/pkg/storage/gcloud_test.go:72]: These tests are not needed anymore?

File: install/installer/third_party/charts/mysql.go [hunk 2]
```diff
@@ -15,7 +15,7 @@ var mysql embed.FS
 
 func MySQL() *Chart {
 	return &Chart{
-		Name:     "MySQL",
+		Name:     "mysql",
```
Comment (felladrin) [inline install/installer/third_party/charts/mysql.go:18]: Changing "name" property has no side effects?

**Review by felladrin — APPROVED** [review]

File: components/content-service/pkg/storage/gcloud_test.go (hunk 1 above)
Comment (aledbf) [reply components/content-service/pkg/storage/gcloud_test.go:72]: Good question. We are removing the upload code (Go) and using `gsutil` binary instead.

**Review by sagor999 — APPROVED** [review]

**Review by vulkoingim — APPROVED** [review]

**Review by corneliusludmann — APPROVED** [review]

**Review by Furisto — APPROVED** [review]

**Pull request merged**